handbook/best_practices.md

```diff
@@ -0,0 +1,152 @@
+
+## Creating secure containers
+
+Creating secure containers is essential to minimize vulnerabilities and ensure the reliability of your applications. Below are some best practices to follow when building container images:
+
+### Prefer rootless containers
+
+Whenever possible, run containers as a non-root user (rootless containers). Rootless containers improve security by reducing the risk of privilege escalation and limiting the impact of potential vulnerabilities. This practice is especially important when deploying containers on NethServer 8 or similar platforms. Ensure that the software you intend to run supports rootless operation, and configure your container runtime accordingly.
+
+### Use multi-stage builds
+
+Multi-stage builds help reduce the size of the final container image by separating the build environment from the runtime environment. This approach ensures that only the necessary artifacts are included in the final image, excluding build tools and dependencies.
+
+For example, when building a Go application:
+```Dockerfile
+# Stage 1: Build
+FROM golang:1.20 AS builder
+WORKDIR /app
+COPY . .
+RUN go build -o myapp
+
+# Stage 2: Runtime
+FROM alpine:3.18
+WORKDIR /app
+COPY --from=builder /app/myapp .
+CMD ["./myapp"]
+```
+
+### Use minimal base images
+
+Whenever possible, use minimal base images like `Alpine` to reduce the attack surface and the number of vulnerabilities. For example:
+
+```Dockerfile
+FROM alpine:3.18
+```
+However, be cautious when using minimal images, as some software may not be fully compatible with them. For instance, MUSL (used in Alpine) can cause issues with certain applications. In such cases, consider using a more compatible base image like `Debian`,
+maybe with the `slim` variant:
+
+```Dockerfile
+FROM debian:12.11-slim
+```
+
+### Pin image versions
+
+Always specify the version of the base image to ensure consistency and avoid unexpected changes. For example:
+```Dockerfile
+FROM alpine:3.18
+```
+Avoid using untagged or `latest` tags, as they can lead to unpredictable builds and runtime issues.
+
+### Consider dependency pinning
+
+When installing packages with managers like `apk` or `apt`, you may choose to pin package versions to improve build reproducibility. For example:
+```Dockerfile
+RUN apk add --no-cache openssl=3.0.9-r1
+```
+However, strict pinning can delay important security updates. Weigh the benefits of reproducibility against the need for timely patches, and choose an approach that fits your project's requirements.
+
+### Automate updates with Renovate
+
+Use tools like [Renovate](https://www.mend.io/renovate/) to automate dependency updates. Renovate can create pull requests for updated dependencies, allowing you to review and merge them as needed. This approach ensures that your images stay up-to-date with minimal manual effort.
```

Comment:
Maybe here can be added a documentation on how to use the global renovate configs?

https://github.com/NethServer/.github